src/Factory/Consignment/ConsignmentFactory.php

```diff
@@ -97,16 +100,31 @@ public function fromOrder(
         $this->setOrderData($order, $deliveryOptions);
         $this->createConsignment();
 
-        $collection        = new ConsignmentCollection();
-        $consignment       = $this->initConsignment();
-        $labelAmount       = $this->request['extraOptions']['labelAmount'] ?? 1;
+        $collection   = new ConsignmentCollection();
+        $extraOptions = new ExtraOptions();
+        $consignment  = $this->initConsignment();
+        $labelAmount  = $this->request['extraOptions']['labelAmount'] ?? 1;
+        $extraOptions->setLabelAmount($labelAmount);
         $distributedWeight = $consignment->getTotalWeight() / $labelAmount;
         $consignment->setTotalWeight($distributedWeight);
 
-        for ($i = 0; $i < $labelAmount; $i++) {
-            $collection->addConsignment($consignment);
+        if ($extraOptions->getLabelAmount() > 1) {
+            $countryService = new CountryService();
+            $orderIso       = $countryService->getShippingCountryIso2($order);
```

Comment:
Als je nog iets aan deze pr aanpast dan ook graag een betere naam voor $orderIso, het is een kortlevende variabele maar dus wel ok, maar als ik ergens $orderIso lees, weet ik niet wat ermee bedoeld zou kunnen worden.